repos/system_upgrade/common/actors/detectcorruptedgrubenv/actor.py

```diff
@@ -0,0 +1,43 @@
+from leapp import reporting
+from leapp.actors import Actor
+from leapp.libraries.common.config import architecture
+from leapp.libraries.actor.detectcorruptedgrubenv import is_grubenv_corrupted
+from leapp.models import GrubConfigError, FirmwareFacts
+from leapp.reporting import create_report, Report
+from leapp.tags import ChecksPhaseTag, IPUWorkflowTag
+
+
+class DetectCorruptedGrubenv(Actor):
+    """
+    Check the grubenv config file has a 1K size and don't end with a line feed
+    """
+
+    name = 'detect_corrupted_grubenv'
+    consumes = (FirmwareFacts,)
+    produces = (Report, GrubConfigError)
+    tags = (ChecksPhaseTag, IPUWorkflowTag)
+
+    def process(self):
+        if architecture.matches_architecture(architecture.ARCH_S390X):
+            return
+        config = '/boot/grub2/grubenv'
+        for fact in self.consume(FirmwareFacts):
+            if fact.firmware == 'efi':
+                config = '/boot/efi/EFI/redhat/grubenv'
+                break
+        if is_grubenv_corrupted(config):
+            create_report([
+                reporting.Title('Detected a corrupted grubenv file.'),
+                reporting.Summary(
+                    'The file %s is missing or corrupted. '
```

Comment:
IMHO not having the file at all should not be considered as an error (proper handling is made in `grub.cfg`).